06_puppy/jimbotech/mapStore_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store  Storer
+	mapper mapCheck
+}
+
+func TestStorer(t *testing.T) {
+	m := NewPuppyStorer()
+	ms := &SyncMapStore{}
+	suite.Run(t, &storesSuite{store: m, mapper: m})
+	suite.Run(t, &storesSuite{store: ms, mapper: ms})
+}
+
+// TestMapStoreWithoutContructor may not be required as it may not be
+// possible to create an instance of that type without using the constructor
+func TestMapStoreWithoutContructor(t *testing.T) {
+	var puppyStore MapStore
+	pup := Puppy{1, "kelpie", "brown", "indispensable"}
+
+	_, err := puppyStore.CreatePuppy(&pup)
+	f := func(t *testing.T) {
+		assert.NotNil(t, err)
+		assert.Equal(t, "store not created, call initializing constructor first", err.Error())
+	}
+	t.Run("CreatePuppy test", f)
+
+	err = puppyStore.UpdatePuppy(1, &pup)
+	t.Run("UpdatePuppy test", f)
+	_, err = puppyStore.ReadPuppy(1)
+	t.Run("ReadPuppy test", f)
+	puppyStore.DeletePuppy(1)
+}
+
+// TestCreateSuccess add to the store and verify
+// by reading that it is in the store and then remove it
+// to leave the store in the original state
+func (s *storesSuite) TestCreateSuccess() {
+	asserter := assert.New(s.T())
```

Comment:
You don't need `asserter` - your `suite` embed an `assert.Assertions` so you can just use `s.Nil` or `s.Equal`, etc. You get all the testify/assert methods.

Reply:
That's what you get if you copy code and not really understand it. Fixed now.